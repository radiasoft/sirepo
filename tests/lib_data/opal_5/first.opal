
OPTION, PSDUMPFREQ = 500;   // 6d data written every 10 time steps (h5).
OPTION, STATDUMPFREQ = 500;  // Beam Stats written every 10 time steps (stat).
OPTION, BOUNDPDESTROYFQ = 10; // Delete lost particles, if any
OPTION, AUTOPHASE = 4;
OPTION, VERSION = 10900;

//----------------------------------------------------------------------------
//Global Parameters
//----------------------------------------------------------------------------

REAL n_particles=41428;      // Number of particles in simulation. (Overridden if distibution imported)

// Initial Momentum (Set very small when starting simulation from cathode emission)
REAL P0 = 1.0e-6;    //inital z momentum
REAL rf_frequency = 1300.0;

REAL gun_phase = -20.0;
REAL cc1_phase = -2.0;
REAL cc2_phase = -10.0;
REAL solenoid_strength_parameter = 0.17765710979201543;
REAL I_Q106 = 0.0;
REAL I_Q107 = 0.0;

REAL beam_current = 0.13;
//----------------------------------------------------------------------------
// Injector Accelerating Cavities
//----------------------------------------------------------------------------


GUN: RFCAVITY, L = 0.265, VOLT = 42.93509225,
    FREQ = rf_frequency,
    ELEMEDGE = 0.0,
    FMAPFN = "GUN_SF7.T7",
    LAG = gun_phase * Pi / 180,
    APVETO = TRUE;

MS:  Solenoid, L = 0, ELEMEDGE = -0.0017, KS = solenoid_strength_parameter,
    FMAPFN = "Should_be_ignored.T7";

DG1: DRIFT, L = 0.363253, ELEMEDGE = GUN->ELEMEDGE + GUN->L;

DG2: DRIFT, L = 0.256055, ELEMEDGE = DG1->ELEMEDGE + DG1->L;

DG3: DRIFT, L = 0.039253, ELEMEDGE = DG2->ELEMEDGE + DG2->L;

DG4: DRIFT, L = 0.262251, ELEMEDGE = DG3->ELEMEDGE + DG3->L;

DG5: DRIFT, L = 0.246295, ELEMEDGE = DG4->ELEMEDGE + DG4->L;

DG6: DRIFT, L = 0.193948, ELEMEDGE = DG5->ELEMEDGE + DG5->L;

DG7: DRIFT, L = 0.076743, ELEMEDGE = DG6->ELEMEDGE + DG6->L;

DG8: DRIFT, L = 0.420191, ELEMEDGE = DG7->ELEMEDGE + DG7->L;

CC1: RFCAVITY, L = 1.3492, VOLT = 28.57233448,
    FREQ = rf_frequency,
    ELEMEDGE = DG8->ELEMEDGE + DG8->L,
    FMAPFN = "TESLA_SF7.T7",
    LAG = cc1_phase * Pi/180;

DCC1_1: DRIFT, L = 0.58952, ELEMEDGE = CC1->ELEMEDGE + CC1->L;

DCC1_2: DRIFT, L = 0.045959, ELEMEDGE = DCC1_1->ELEMEDGE + DCC1_1->L;

DCC1_3: DRIFT, L = 0.724255, ELEMEDGE = DCC1_2->ELEMEDGE + DCC1_2->L;

CC2: RFCAVITY, L = 1.3492, VOLT = 30.6427935,
    FREQ = rf_frequency,
    ELEMEDGE = DCC1_3->ELEMEDGE + DCC1_3->L,
    FMAPFN = "TESLA_SF7.T7",
    LAG = cc2_phase * Pi/180;

DCC2_1: DRIFT, L = 0.801182, ELEMEDGE = CC2->ELEMEDGE + CC1->L;

DCC2_2: DRIFT, L = 0.090742, ELEMEDGE = DCC2_1->ELEMEDGE + DCC2_1->L;

DCC2_3: DRIFT, L = 0.213827, ELEMEDGE = DCC2_2->ELEMEDGE + DCC2_2->L;

DCC2_4: DRIFT, L = 0.120424, ELEMEDGE = DCC2_3->ELEMEDGE + DCC2_3->L;

DCC2_5: DRIFT, L = 0.343, ELEMEDGE = DCC2_4->ELEMEDGE + DCC2_4->L;

DCC2_6: DRIFT, L = 0.343, ELEMEDGE = DCC2_5->ELEMEDGE + DCC2_5->L;

DCC2_7: DRIFT, L = 0.32599, ELEMEDGE = DCC2_6->ELEMEDGE + DCC2_6->L;

DCC2_8: DRIFT, L = 0.078595, ELEMEDGE = DCC2_7->ELEMEDGE + DCC2_7->L;

DCC2_9: DRIFT, L = 0.126538, ELEMEDGE = DCC2_8->ELEMEDGE + DCC2_8->L;

DCC2_10: DRIFT, L = 0.078569, ELEMEDGE = DCC2_9->ELEMEDGE + DCC2_9->L;

X_106: MONITOR, OUTFN = 'output_x_106.h5', ELEMEDGE = DCC2_10->ELEMEDGE + DCC2_10->L;

Q_106: QUADRUPOLE, L = 0.167, ELEMEDGE = X_106->ELEMEDGE, K1S = I_Q106 * 10.135 * 40 / ( 1.8205 * 299.8 );

DQ106: DRIFT, L = 0.034249, ELEMEDGE = Q_106->ELEMEDGE + Q_106->L;

Q_107: QUADRUPOLE, L = 0.167, ELEMEDGE = DQ106->ELEMEDGE + DQ106->L, K1S = I_Q107 * 10.135 * 40 / ( 1.8205 * 299.8 );

DX107: DRIFT, L = 0.255184, ELEMEDGE = Q_107->ELEMEDGE + Q_107->L;

X_107: MONITOR, OUTFN = 'output_x_107.h5', ELEMEDGE = DX107->ELEMEDGE + DX107->L;

LINAC: LINE = (GUN, MS, CC1, CC2, X_106, Q_106, Q_107, X_107);

// Still needs adjustment, bunch length does not match very well.
REAL fwhm = 3.12 / 360 * (1 / (rf_frequency*1.0e6));

gen_dist: DISTRIBUTION, TYPE = FLATTOP,
        SIGMAR = 0.001*2,
        TPULSEFWHM = fwhm*2,
        NBIN = 9,
        EMISSIONSTEPS = 100,
        EMISSIONMODEL = NONE,
        EKIN = 0.55,
        EMITTED = True,
        WRITETOFILE = True;    //Saves the distribution to a text file

vc_dist: DISTRIBUTION, TYPE = FROMFILE,
        FNAME = "fast_laser.dist",
        EMISSIONMODEL = None,
        NBIN = 9,
        EMISSIONSTEPS = 100,
        EKIN = 0.4,
        EMITTED = True;

//----------------------------------------------------------------------------
// Define Field solvers
//----------------------------------------------------------------------------

FS_SC: Fieldsolver, FSTYPE = FFT, // None or FFT
            MX = 26, MY = 26, MT = 25,
            PARFFTX = false,
            PARFFTY = false,
            PARFFTT = true,  //parallel in the z direction only
            BCFFTX = open,
            BCFFTY = open,
            BCFFTT = open,
            BBOXINCR = 1,
            GREENSF = INTEGRATED;

//----------------------------------------------------------------------------
// Electron Beam Definition

BEAM1:  BEAM, PARTICLE = ELECTRON, pc = P0, NPART = n_particles,
        BFREQ = 1300.0, BCURRENT = beam_current, CHARGE = -1;

//----------------------------------------------------------------------------
// Initialize Simulation
//----------------------------------------------------------------------------


TRACK, LINE = LINAC, BEAM = BEAM1, MAXSTEPS = 2500000,
    DT = {2.0e-13}, ZSTOP={9.25};

RUN, METHOD = "PARALLEL-T", BEAM = BEAM1,
    FIELDSOLVER = FS_SC, DISTRIBUTION = vc_dist;
ENDTRACK;

Stop;
Quit;
