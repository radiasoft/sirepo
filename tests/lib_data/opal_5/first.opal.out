
Title, string="first";

REAL beam_current = 0.13;
REAL cc1.elemedge = 2.1229889999999996;
REAL cc1.l = 1.3492;
REAL cc1_phase = -2.0;
REAL cc2.elemedge = 4.831923;
REAL cc2_phase = -10.0;
REAL dcc1_1.elemedge = 3.4721889999999993;
REAL dcc1_1.l = 0.58952;
REAL dcc1_2.elemedge = 4.061709;
REAL dcc1_2.l = 0.045959;
REAL dcc1_3.elemedge = 4.107667999999999;
REAL dcc1_3.l = 0.724255;
REAL dcc2_1.elemedge = 6.1811229999999995;
REAL dcc2_1.l = 0.801182;
REAL dcc2_10.elemedge = 8.624420999999998;
REAL dcc2_10.l = 0.078569;
REAL dcc2_2.elemedge = 6.982304999999999;
REAL dcc2_2.l = 0.090742;
REAL dcc2_3.elemedge = 7.073046999999999;
REAL dcc2_3.l = 0.213827;
REAL dcc2_4.elemedge = 7.286873999999999;
REAL dcc2_4.l = 0.120424;
REAL dcc2_5.elemedge = 7.407297999999999;
REAL dcc2_5.l = 0.343;
REAL dcc2_6.elemedge = 7.750297999999999;
REAL dcc2_6.l = 0.343;
REAL dcc2_7.elemedge = 8.093297999999999;
REAL dcc2_7.l = 0.32599;
REAL dcc2_8.elemedge = 8.419287999999998;
REAL dcc2_8.l = 0.078595;
REAL dcc2_9.elemedge = 8.497882999999998;
REAL dcc2_9.l = 0.126538;
REAL dg1.elemedge = 0.265;
REAL dg1.l = 0.363253;
REAL dg2.elemedge = 0.628253;
REAL dg2.l = 0.256055;
REAL dg3.elemedge = 0.8843079999999999;
REAL dg3.l = 0.039253;
REAL dg4.elemedge = 0.9235609999999999;
REAL dg4.l = 0.262251;
REAL dg5.elemedge = 1.1858119999999999;
REAL dg5.l = 0.246295;
REAL dg6.elemedge = 1.4321069999999998;
REAL dg6.l = 0.193948;
REAL dg7.elemedge = 1.6260549999999998;
REAL dg7.l = 0.076743;
REAL dg8.elemedge = 1.7027979999999998;
REAL dg8.l = 0.420191;
REAL dq106.elemedge = 8.869989999999998;
REAL dq106.l = 0.034249;
REAL dx107.elemedge = 9.071238999999998;
REAL dx107.l = 0.255184;
REAL fwhm = 6.666666666666666e-12;
REAL gun.elemedge = 0.0;
REAL gun.l = 0.265;
REAL gun_phase = -20.0;
REAL i_q106 = 0.0;
REAL i_q107 = 0.0;
REAL n_particles = 41428.0;
REAL p0 = 1e-06;
REAL q_106.elemedge = 8.702989999999998;
REAL q_106.l = 0.167;
REAL q_107.elemedge = 8.904238999999999;
REAL q_107.l = 0.167;
REAL rf_frequency = 1300.0;
REAL solenoid_strength_parameter = 0.17765710979201543;
REAL x_106.elemedge = 8.702989999999998;


"OPT1": option,autophase=4.0,psdumpfreq=500.0,statdumpfreq=500.0,version=10900.0;

"X_106": MONITOR,outfn="output_x_106.h5";
"X_107": MONITOR,outfn="output_x_107.h5";
"Q_106": QUADRUPOLE,l=0.167;
"Q_107": QUADRUPOLE,l=0.167;
"CC1": RFCAVITY,fmapfn="TESLA_SF7.T7",freq=1300.0,l=1.3492,lag=-0.03490658503988659,volt=28.57233448;
"CC2": RFCAVITY,fmapfn="TESLA_SF7.T7",freq=1300.0,l=1.3492,lag=-0.17453292519943295,volt=30.6427935;
"GUN": RFCAVITY,apveto=true,fmapfn="GUN_SF7.T7",freq=1300.0,l=0.265,lag=-0.3490658503988659,volt=42.93509225;
"MS": SOLENOID,fmapfn="SOLENO4.T7",ks=0.17765710979201543;

"MS#0": "MS",elemedge=-0.0017;
"GUN#0": "GUN",elemedge=0.0;
"CC1#0": "CC1",elemedge=2.1229889999999996;
"CC2#0": "CC2",elemedge=4.831923;
"X_106#0": "X_106",elemedge=8.702989999999998;
"Q_106#0": "Q_106",elemedge=8.702989999999998;
"Q_107#0": "Q_107",elemedge=8.904238999999999;
"X_107#0": "X_107",elemedge=9.326422999999998;
LINAC: LINE=("MS#0","GUN#0","CC1#0","CC2#0","X_106#0","Q_106#0","Q_107#0","X_107#0");


"GEN_DIST": distribution,ekin=0.55,emissionsteps=100.0,emitted=true,nbin=9.0,sigmar=0.002,tpulsefwhm=1.3333333333333333e-11,type="FLATTOP",writetofile=true;
"VC_DIST": distribution,ekin=0.4,emissionsteps=100.0,emitted=true,fname="fast_laser.dist",nbin=9.0,type="FROMFILE";
"FS_SC": fieldsolver,bboxincr=1.0,fstype="FFT",mt=25.0,mx=26.0,my=26.0;
"BEAM1": beam,bcurrent=0.13,bfreq=1300.0,charge=-1.0,npart=41428.0,particle="ELECTRON",pc=1e-06;
"TR1": track,beam=BEAM1,dt={2.0e-13},line=LINAC,maxsteps=2500000.0,zstop={9.25};
 run, beam=BEAM1,distribution=vc_dist,fieldsolver=FS_SC,method="PARALLEL-T";
endtrack;
